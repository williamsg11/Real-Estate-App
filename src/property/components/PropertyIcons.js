import React, {Component, PropTypes} from 'react';
import {StyleSheet, Text, View} from 'react-native';
import FontAwesome from 'react-native-vector-icons/FontAwesome';
import colors from '../../common/colors';

export default class PropertyIcons extends Component {
  static propTypes = {
    services: PropTypes.object.isRequired,
    items: PropTypes.array.isRequired,
  };

  renderIcon(value, icon, index) {
    return (
      <View key={index} style={styles.row}>
        <FontAwesome name={icon} size={15} color={colors.fadedBlack} />
        <Text
          style={{
            fontSize: 13,
            fontWeight: '100',
            marginHorizontal: 2,
            color: colors.black,
          }}>
          {value}
        </Text>
      </View>
    );
  }

  render() {
    const {services, items} = this.props;
    return (
      <View style={styles.container}>
        {items.map((item, i) => {
          switch (item) {
            case 'bedroom':
              return this.renderIcon(services[item], 'bed', i);
            case 'bathroom':
              return this.renderIcon(services[item], 'bath', i);
            case 'parking':
              return this.renderIcon(services[item], 'car', i);
            default:
              return null;
          }
        })}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    paddingVertical: 2,
  },
  row: {
    flexDirection: 'row',
    paddingLeft: 0,
    paddingRight: 10,
    alignItems: 'center',
  },
  title: {
    color: 'white',
    fontSize: 15,
    paddingLeft: 5,
  },
  icon: {
    width: 20,
    height: 25,
    color: 'white',
  },
});
