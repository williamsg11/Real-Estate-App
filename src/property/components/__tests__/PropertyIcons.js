import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import PropertyIcons from './../PropertyIcons';

test('renders heart', () => {
  const tree = renderer
    .create(
      <PropertyIcons
        services={{bedroom: '1', bathroom: '2'}}
        items={['bedroom']}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
