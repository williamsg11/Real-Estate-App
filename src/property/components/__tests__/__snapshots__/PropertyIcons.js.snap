// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders heart 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "paddingVertical": 2,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingLeft": 0,
        "paddingRight": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "rgba(0, 0, 0, .5)",
            "fontSize": 15,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      ïˆ¶
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000",
          "fontSize": 13,
          "fontWeight": "100",
          "marginHorizontal": 2,
        }
      }
    >
      1
    </Text>
  </View>
</View>
`;
